core/src/main/java/com/yahoo/oak/ReferenceCodecNova.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+/**
+ * A reference that is involved in synchronized and recycling memory management is composed of
+ * 3 parameters:block ID, offset and version.
+ *
+ * 0...            ...42 | 43... ...62 | 63
+ * block ID   |  offset  | version     | deleted 
+ * first         second      third
+ *
+ * All these parameters may be squashed together into one long for easy representation.
+ * Using different number of bits for each parameter may incur different limitations on their sizes.
+ */
+class ReferenceCodecNova extends ReferenceCodec {
+
+    static final int    INVALID_VERSION = 0;
+
+    // All Oak instances on one machine are expected to reference to no more than 4TB of RAM.
+    // 4TB = 2^42 bytes
+    // blockIDBitSize + offsetBitSize = BITS_FOR_MAXIMUM_RAM
+    // The number of bits required to represent such memory:
+    private static final int BITS_FOR_MAXIMUM_RAM = 42;
+    private static final long VERSION_DELETE_BIT_MASK = (1 << (Long.SIZE - Long.SIZE));
```

Comment:
Why there is a difference (in `-1`) from definition of `VERSION_DELETE_BIT_MASK` in `ReferenceCodecSyncRecycle` is it a bug there?